python/cudf_polars/cudf_polars/experimental/shuffle.py

```diff
@@ -192,9 +233,47 @@ def _(
 def _(
     ir: Shuffle, partition_info: MutableMapping[IR, PartitionInfo]
 ) -> MutableMapping[Any, Any]:
-    # Use a simple all-to-all shuffle graph.
+    # Extract "shuffle_method" configuration
+    if (
+        shuffle_method := ir.config_options.get(
+            "executor_options.shuffle_method",
+            default=None,
+        )
+    ) not in (*_SHUFFLE_METHODS, None):
+        raise ValueError(
+            f"{shuffle_method} is not a supported shuffle method. "
+            f"Expected one of: {_SHUFFLE_METHODS}."
+        )
+
+    # Try using rapidsmp shuffler if we have "simple" shuffle
+    # keys, and the "shuffle_method" config is set to "rapidsmp"
+    _keys: list[Col]
+    if shuffle_method in (None, "rapidsmp") and len(
+        _keys := [ne.value for ne in ir.keys if isinstance(ne.value, Col)]
+    ) == len(ir.keys):  # pragma: no cover
+        shuffle_on = [k.name for k in _keys]
+        try:
+            from rapidsmp.integrations.dask import rapidsmp_shuffle_graph
+
+            return rapidsmp_shuffle_graph(
+                get_key_name(ir.children[0]),
+                get_key_name(ir),
+                list(ir.schema.keys()),
```

Comment:
Down below, we use `ir.keys`. Just confirming that we really do want `ir.schema.keys()` here.

Reply:
Also noticing that `_simple_shuffle_graph` uses `name_out, name_in` as the first two parameters, while `rapidsmp_shuffle_graph` uses `name_in, name_out` :)

Not worth changing immediately, maybe if/when we move `rapidsmp_shuffle_graph`?

Reply:
>Down below, we use ir.keys. Just confirming that we really do want ir.schema.keys() here.

Yeah, good question. The problem is that the rapidsmp shuffler doesn't support shuffling on general expressions yet (only a list of column names). This is because we are using `rapidsmp.shuffler.partition_and_pack` to partition the data. We may not *need* to use that function anymore, but we definitely **did** need to use it to get effective spilling at one point.

>Not worth changing immediately, maybe if/when we move rapidsmp_shuffle_graph?

Heh - Good point. I like the order used in `rapidsmp_shuffle_graph`, so I just changed `_simple_shuffle_graph`.